src/Norm/NormGUI/src/normGUI/engine/GUIDoubleElement.java

```diff
@@ -106,10 +106,10 @@ public Object getValueIn(GUIUnit unit)
                 // Appel de la methode native
                 // Remarque: on utilise Float plutot que Double pour limiter le nombre
                 // de chiffres apres la virgule a l'affichage
-                // C'est le seul endroit on on accede a la methode getDoubleValueAt, mais cela
-                // reste une solution "bricolo" pour controler le formatage de l'afficahge des
-                // reels
-                return new Float(unit.getDoubleValueAt(getIdentifier()));
+                // C'est le seul endroit on on accede a la methode getDoubleValueAt,
+                //  mais cela reste une solution "bricolo" pour controler le formatage
+                // de l'affichage des reels
+                return Float.valueOf((float)unit.getDoubleValueAt(getIdentifier()));
```

Comment:
I suppose there is no way to avoid the `Double` -> `Float` cast here? If so, how do we guarantee that there is no information loss?

Reply:
La perte d'information portant sur le nombre de décimales (de 15 à 7) est ici justement ce qui est souhaité
La perte d'information portant sur les risques d'overflow (valeurs > 10^37) est plus qu'improbable, puisque ce besoin de si grandes valeurs à l'interface est inexistant.
En l'occurrence, tous les paramètres de type double actuellement utilisés dans la GUI sont spécifiés avec des bornes [0, 1].
Il n'y a donc pas lieu d'améliorer ce code (compromis coût-bénéfice).